### PR TITLE
Add PHP7 and MongoDB package support

### PR DESCRIPTION
**Pull request by connorcook**
Adds a small package tweak to the work Jose Luis Falcon did so it can be merged back into the original repository without breaking packages.

### CODE REVIEW

File: lib/Store/Store.php [hunk 1]
```diff
@@ -50,14 +57,14 @@ static function createConnectionURI($connectionDetails = array()) {
...
-        if(!empty($connectionDetails['replicaSet'])) {
+            ."${seedList}";
+           // ."/${connectionDetails['database']}";
+        /*if(!empty($connectionDetails['replicaSet'])) {
             $connectionURI .= "?replicaSet=${connectionDetails['replicaSet']}";
             if(!empty($connectionDetails['readPreference'])) {
                 $connectionURI .= "&readPreference=${connectionDetails['readPreference']}";
             }
-        }
+        }*/
```
Comment (ppatriotis) [inline lib/Store/Store.php]: Commented code in PR ;)

File: lib/Store/Store.php [hunk 2]
```diff
@@ -75,28 +82,36 @@ public function get($type, $key)
...
+        $where = [
+        	'session_id' => $key
+        ];
+        $query = new Query($where, ['limit' => 1]);
+
+        $namespace = "{$this->dbName}.{$type}";
+
+        $cursor = $this->manager->executeQuery($namespace, $query);
+
+	    if (false === ($cursor = current($cursor->toArray()))) {
```
Comment (ppatriotis) [inline lib/Store/Store.php]: Looks like some formatting problem

File: lib/Store/Store.php [hunk 3]
```diff
@@ -75,28 +82,36 @@ public function get($type, $key)
...
-        $collection = $this->db->{$type};
-        $document = $collection->findOne(array(
-            'session_id' => $key
-        ));
+        $where = [
+        	'session_id' => $key
+        ];
+        $query = new Query($where, ['limit' => 1]);
+
+        $namespace = "{$this->dbName}.{$type}";
```
Comment (ppatriotis) [inline lib/Store/Store.php]: Don’t see namespace query or where used anywhere else

**Comment by connorcook** [comment]
Haha, @ppatriotis I didn't think we'd see you around these parts again ;P

It's clear I'll need to tweak the work Jose did rather than straight up pulling it in. Gonna do it later, then, still got the Laravel Upgrade to finish first.

... and don't scare me like that, haha

**Comment by connorcook** [comment]
@prolificchrisb I've fixed all the tests and addressed the feedback. It should be good to go, and it is now fully backwards compatible with PHP 5.5-5.6 as well as PHP7.0-7.2.

**Comment by prolificchrisb** [comment]
@connorcook I think we should create a separate repo, SimplesSAMLphp module, and Packagist package for mongodb and leave this repo targeted to the mongo extension.

**Comment by connorcook** [comment]
> @connorcook I think we should create a separate repo, SimplesSAMLphp module, and Packagist package for mongodb and leave this repo targeted to the mongo extension.

Sounds good.

**Pull request closed without merging**